src/cirrus/lib/events.py

```diff
@@ -66,6 +115,372 @@ def sns_attributes(self: Self) -> dict[str, dict[str, str]]:
+            if self.metric_some_workflows not in self._metrics or (
+                self.metric_all_workflows not in self._metrics
+            ):
+                raise ValueError(
+                    f"No metrics found in namespace {self.metric_namespace}",
+                )
+
+    def enabled(self) -> bool:
+        return self._enabled
+
+    def aggregated_for_specified_workflows(
+        self,
+        workflows: list[str],
+        start_time: datetime,
+        end_time: datetime,
+        period: int = 3600,
+        event_types: list[WFEventType] | None = None,
+        statistics: list[str] | None = None,
+        formatter: Callable[[datetime], str] | None = None,
+    ) -> list[dict[str, str | dict[str, dict[str, float]]]]:
+        """
+        Retrieve metric statistics from CloudWatch.
+
+        Args:
+            event_types (list[WFEventType]): List of workflow event types to filter.
+            workflows (list[str]): List of workflow names to filter.
+            start_time (datetime): Start of the time window.
+            end_time (datetime): End of the time window.
+            period (int): Granularity in seconds.
+            statistics (list[str]): List of statistics to retrieve.
+
+        Returns:
+            list[dict[str, Any]]: List of metric statistics dictionaries retrieved.
+        """
+        if statistics is None:
+            statistics = [WorkflowMetricReader._agg_statistic]
+        if event_types is None:
+            event_types = list(WFEventType)
+        if formatter is None:
+            formatter = date_formatter()
+        delta = timedelta(seconds=period)
+        dates = [
+            start_time + i * delta
+            for i in range(int((end_time - start_time).total_seconds() / period))
+        ]
+        fcstats: dict[datetime, dict[str, dict[str, float]]] = {
+            d.replace(second=0, microsecond=0): {
+                e: dict.fromkeys(statistics, 0) for e in event_types
+            }
+            for d in dates
+        }
+
+        for workflow in workflows:
+            for event_type in event_types:
+                resp = self.cw_client.get_metric_statistics(
+                    Namespace=self.metric_namespace,
+                    MetricName=self.metric_some_workflows,
+                    Dimensions=[
+                        {"Name": "event", "Value": str(event_type)},
+                        {"Name": "workflow", "Value": workflow},
+                    ],
+                    StartTime=start_time,
+                    EndTime=end_time,
+                    Period=period,
+                    Statistics=statistics,
+                )
+                for datapoint in resp["Datapoints"]:
+                    tstamp = datapoint["Timestamp"]
+                    for statistic in statistics:
+                        stat = datapoint[statistic]
+                        fcstats[tstamp][str(event_type)][statistic] += stat
+        # TODO: break this out by workflow, maybe
```

Comment:
Does this still apply? More context on why to consider this might be good.

Reply:
Actually this whole function needs much updating, and consideration if it is of value.  The `EventDB` did not provide this function, and the `api` lambda went straight to the `StateDB` to summarize each workflow's **_current_** state.  This function would be if we wanted to get temporal statistics for an individual workflow.  
```suggestion
```